app/scripts/metamask-controller.js

```diff
@@ -1398,6 +1399,19 @@ export default class MetamaskController extends EventEmitter {
         allowedActions: [`${this.snapController.name}:handleRequest`],
       }),
     });
+    this.userStorageController = new UserStorageController({
+      messenger: this.controllerMessenger.getRestricted({
+        name: 'UserStorageController',
+        allowedActions: [`${this.snapController.name}:handleRequest`],
+      }),
+      auth: {
+        getBearerToken: () => this.authenticationController.getBearerToken(),
+        getSessionIdentifier: async () =>
```

Comment:
Only one remaining issue from me on this PR. Can we update this to use the ControllerMessenger actions pattern? You can search in this file for `allowedActions` for examples of other controllers that follow this pattern. I am also looking for more documentation on it

Reply:
Great comment, using the Messaging System now!